virtwho/virt/rhevm/rhevm.py

```diff
@@ -71,18 +71,72 @@ def __init__(self, logger, config):
         self.username = self.config.username
         self.password = self.config.password
 
-        self.clusters_url = urlparse.urljoin(self.url, "api/clusters")
-        self.hosts_url = urlparse.urljoin(self.url, "api/hosts")
-        self.vms_url = urlparse.urljoin(self.url, "api/vms")
-
         self.auth = HTTPBasicAuth(self.config.username, self.config.password)
+        self.prepared = False
+
+    def prepare(self):
+        if not self.prepared:
+            if not hasattr(self, 'major_version'):
+                self.getVersion()
+            self.buildUrls()
+            self.prepared = True
+
+    def buildUrls(self):
+        """
+        Builds the URL's based on Rhev version
+        """
+        clusters_endpoint = 'clusters'
+        hosts_endpoint = 'hosts'
+        vms_endpoint = 'vms'
+
+        if self.major_version == "4":
+            api_base = 'ovirt-engine/api/'
+        else:
+            api_base = 'api/'
+
+        self.clusters_url = urlparse.urljoin(self.url, api_base + clusters_endpoint)
+        self.hosts_url = urlparse.urljoin(self.url, api_base + hosts_endpoint)
+        self.vms_url = urlparse.urljoin(self.url, api_base + vms_endpoint)
+
+    def getVersion(self):
```

Comment:
Can we change the new method names to pep8 standard of lowercase separated by '_' where necessary?
(e.g. "get_version")

We can make a separate task to go back through the code base and perform this change on the rest of the functions and methods.